src/base-filter.ts

```diff
@@ -5,7 +5,7 @@ import Options, { OptionsInterface } from './options'
  * Filter base class
  */
 export default abstract class BaseFilter {
-  protected readonly name: string
+  protected name: string
```

Comment:
Actually you only need to make this property public and non-readonly, no need to add `filterName` getter/setter.

Reply:
Yes, for this simple purpose(change property **name**), make it public and non-readonly also works, but for OOP, directly access a property of class sometime not a good idea, by getter/setter, wrong value can be intercepted and inform user such error.
Anyway, I think either directly access or getter/setter are ok. You are the owner of rebloom, you can make a decision.

Reply:
That's fine, at least for now, I think.
In this PR, `filterName` is nothing more than an alias to `name`. We don't validate or transform input or something else.
In the future, if really need, we can make `name` getter/setter to another private property.